lib/src/components/NxLoadError/__tests__/NxLoadError.test.tsx

```diff
@@ -4,86 +4,99 @@
-    expect(getShallowComponent({ error: 'Error!' })).not.toContainMatchingElement(NxButton);
-
-    expect(getShallowComponent({ error: 'Error!', retryHandler: () => {} })).toContainMatchingElement(NxButton);
-    expect(getShallowComponent({ error: 'Error!', retryHandler: () => {} }).find(NxButton)).toIncludeText('Retry');
+    const elWithoutRetryButton = renderEl({ error: 'Error' });
+    expect(elWithoutRetryButton?.textContent).not.toContain('Retry');
+    const elRetryButton = renderEl({ error: 'Error', retryHandler: () => {} });
+    const retryButton = screen.getByRole('button', {name: 'Retry'});
+    expect(elRetryButton?.textContent).toContain('An error occurred loading data. Error');
+    expect(retryButton).toBeInTheDocument();
   });
 
   it('adds the appropriate class, variant, and type to the retry button', function() {
-    const button = getShallowComponent({ error: 'Error!', retryHandler: () => {} }).find(NxButton);
-
-    expect(button).toHaveClassName('nx-load-error__retry');
-    expect(button).toHaveProp('variant', 'error');
-    expect(button).toHaveProp('type', 'button');
+    renderEl({ error: 'Error', retryHandler: () => {} });
+    const retryButton = screen.getByRole('button', {name: 'Retry'});
+    expect(retryButton).toBeInTheDocument();
+    expect(retryButton).toHaveClass('nx-load-error__retry');
+    expect(retryButton).toHaveAttribute('type', 'button');
   });
 
   it('sets the retry button type to "submit" when submitOnRetry is set to true', function() {
-    const props = { error: 'Error!', submitOnRetry: true };
-    const button = getShallowComponent(props).find(NxButton);
-    expect(button).toHaveProp('type', 'submit');
+    renderEl({ error: 'Error', submitOnRetry: true });
+    const retryButton = screen.getByRole('button', {name: 'Retry'});
+    expect(retryButton).toBeInTheDocument();
+    expect(retryButton).toHaveAttribute('type', 'submit');
   });
 
-  it('calls the retryHandler when the retry button is clicked', function() {
-    const retryHandler = jest.fn(),
-        props = { error: 'Error!', canRetry: true, retryHandler },
-        component = getShallowComponent(props);
-
+  it('calls the retryHandler when the retry button is clicked', async function() {
+    const user = userEvent.setup();
+    const retryHandler = jest.fn();
+    renderEl({ error: 'Error!', retryHandler });
     expect(retryHandler).not.toHaveBeenCalled();
-
-    component.find(NxButton).simulate('click');
-
+    const retryButton = screen.getByRole('button', {name: 'Retry'});
+    expect(retryButton).toBeInTheDocument();
+    await user.click(retryButton);
     expect(retryHandler).toHaveBeenCalled();
   });
 
-  it('passes unknown props to the NxErrorAlert element', function() {
-    const onClick = jest.fn(),
-        component = getShallowComponent({ error: 'err', id: 'foo', onClick });
+  it('calls the onClose function when the "X" button is clicked (to dismiss alert)', async function() {
```

Comment:
I'd break this down a bit.  I'd create a `describe` block for "when onClose is defined" and they I'd have two tests: one that tests that that causes the Close button to be rendered (and that the Close button is not rendered when `onClose` is not defined), and a second test that tests that clicking the Close button fires `onClose`

Reply:
> when onClose is defined

Done 9c3a7b928eb032da2bbd061cecf1ba01a191e71b